### PR TITLE
Standardise command suggestion formatting

### LINKED ISSUE #2225: Change docs command suggestions to follow style convention
## Description
As part of #2014 it was discovered that we are not consistent in how we format cli commands with flags. It was decided that we should be using `kedro command --flag=value1,value2` and `kedro command --flag='value with spaces 1, value with spaces 2'`.

This will affect: 
https://kedro.readthedocs.io/en/stable/development/commands_reference.html#modifying-a-kedro-run
https://kedro.readthedocs.io/en/stable/data/data_catalog.html#version-datasets-and-ml-models
https://kedro.readthedocs.io/en/stable/kedro_project_setup/configuration.html#specify-parameters-at-runtime
https://kedro.readthedocs.io/en/stable/get_started/new_project.html#create-a-new-project-from-a-configuration-file
https://kedro.readthedocs.io/en/stable/visualisation/kedro-viz_visualisation.html
https://kedro.readthedocs.io/en/stable/nodes_and_pipelines/run_a_pipeline.html#run-a-pipeline-by-name
https://kedro.readthedocs.io/en/stable/nodes_and_pipelines/slice_a_pipeline.html
https://kedro.readthedocs.io/en/stable/kedro_project_setup/configuration.html#configure-kedro-run-arguments
https://kedro.readthedocs.io/en/stable/deployment/dask.html

(This list may be incomplete)

Note: https://kedro.readthedocs.io/en/stable/development/commands_reference.html#modifying-a-kedro-run has a list of flags to modify `kedro run`. This list is missing `--conf-source`, `--to-outputs`, and `--async`. These should be added in (with the same formatting convention).

### PR DESCRIPTION
**Pull request by AhdraMeraliQB**
Signed-off-by: Ahdra Merali <ahdra.merali@quantumblack.com>

> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.

## Description
Resolves #2225 

## Development notes

- [x] Standardise command suggestion formatting
- [x] Confirm changed commands still works as expected
- [x] Add missing commands `--to-outputs`, and `--async`
- [x] Add formatting to style guide (if not already there)
- [x] Build docs and test pages render as expected

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/2253"></a>

### CODE REVIEW

**Comment by stichbury** [comment]
This looks great to me @AhdraMeraliQB but I will build the docs in your branch and check them locally to see how the table is looking, and if there's any improvement we can do for readability on it.

**Comment by stichbury** [comment]
I'm seeing this table: 
image

Was this the ticket where you were removing the third column or will that happen separately?

I wonder if it's worth removing anyway, and potentially just adding "Multiple instances allowed" for the very few rows that answer "Yes" at present??

**Comment by AhdraMeraliQB** [comment]
@stichbury 
> Was this the ticket where you were removing the third column or will that happen separately?

I believe you're thinking of [this issue](https://github.com/kedro-org/kedro/issues/2246). That's to be done once the implementation allowing multiple instances is removed, and is handled separately as the change would affect other aspects of the documentation as well.

**Comment by stichbury** [comment]
Makes sense @AhdraMeraliQB -- thanks! I think though that it would make sense to improve the table usability now so I made the changes on your branch, which build to the resulting table:

image


If you agree and are happy, I can commit my change in so you pick up the new table, but if you'd prefer to leave as is until the other issue is resolved, let's do that. Either way, I've had my day's practice at editing markdown tables :)

**Comment by AhdraMeraliQB** [comment]
@stichbury I like the changes - definitely makes more sense instead of having a column that applies to only three cases, and the later ticket can just remove those extra lines. I'm happy to go with your suggestion here. 

Just a note - `--params` having multiple instances allowed is actually an error in the docs, `kedro run` will not recognise multiple params flags.

File: docs/source/data/data_catalog.md [hunk 1]
```diff
@@ -509,18 +509,18 @@ Making a simple addition to your Data Catalog allows you to perform versioning o
 Consider the following versioned dataset defined in the `catalog.yml`:
 
 ```yaml
-cars.csv:
```
Comment (AhdraMeraliQB) [inline docs/source/data/data_catalog.md:512]: Noticed that the naming here was inconsistent with other data catalog examples, including others on the same page. Dropped the file extension to fix that.

**Review by ankatiyar — APPROVED** [review]
👍

**Review by stichbury — APPROVED** [review]
LGTM 🏆 💨

**Pull request merged**